supervisor/qemu/src/main.rs

```diff
@@ -989,6 +999,47 @@ impl QemuSupervisor {
             qemu_proc,
         });
     }
+
+    async fn qemu_monitor_shutdown(qmp_socket_path: &str) -> Result<(), anyhow::Error> {
```

Comment:
This should either use a `PathBuf`, `std::os::unix::net::SocketAddr`, or `std::net::SocketAddr`.